src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts

```diff
@@ -135,6 +153,30 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
     this.urlParamsSubscription?.unsubscribe();
   }
 
+  isScoresColumn(column: string) {
+    const isScore = this.scoresColumns.find(c => c.field === column);
+    return isScore !== undefined ? true : false;
+  }
+
+  toggleGCTColumn(column: GCTColumn) {
+    column.selected = !column.selected;
+    this.updateVisibleColumns();
+    this.onResize();
+  }
+
+  updateVisibleColumns() {
+    const visibleScoresColumns: string[] = 
+      this.scoresColumns
```

Comment:
...`.filter(c=> c.visible && c.selected).map`... should be sufficient and same below